master/buildbot/www/hooks/github.py

```diff
@@ -233,6 +346,143 @@ def _process_change(self, payload, user, repo, repo_url, project, event,
 
         return changes
 
+    @defer.inlineCallbacks
+    def _get_http_client(self):
+        if self._http is None:
+            # Instanticate an http client service to user
+            self._http = yield httpclientservice.HTTPClientService.getService(
+                self.master, self.github_api_endpoint, headers={
+                    'Authorization': 'token ' + self.oauth_token,
+                    'User-Agent': 'Buildbot',
+                    'Accept': 'application/vnd.github.v3+json'
+                },
+                debug=self.debug, verify=self.verify)
+            # Get rate limits
+            response = yield self._http.get('/rate_limit')
```

Comment:
This initial rate query is not really necessary but allows us to know at start how we are on our rate limit impositions in case of trouble. It can be removed, just say to remove it.